### PR TITLE
added api hooks for all current backend routes

### PR DESCRIPTION
**Pull request by roccopolimen**
Created all the api calls

The ones that need to provide a body might not work as intended, but I do not have the infrastructure to fully test them yet so I will just modify them if needed if they do not work as expected.

Once this is merged, all front-end components should start using these hooks.

Also I made it the default for the query to NOT call the api call on component mount, so if you want it to do so, just use a useEffect on the refetch function.

### CODE REVIEW

File: client/src/api/routes/applications.ts [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+ * @param {string} name 
+ * @param {string} pronouns 
+ * @param {string} location 
+ * @param {string} phone 
+ * @param {string} email 
+ * @returns {UseQueryResult<ContactObject>} the newly created contact
+ * @throws if fails
+ */
+export const useCreateContact = (applicationId: string, name: string, pronouns: string, location: string, phone: string, email: string): UseQueryResult<ContactObject> => {
+    return useQuery('createEvent', async () => {
```
Comment (bwormser) [inline client/src/api/routes/applications.ts]: Should this be createEvent or createContact?

File: client/src/api/routes/applications.ts [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+/**
+ * @description DELETE /application/:applicationId
+ * @param {string} applicationId 
+ * @returns {UseQueryResult<boolean>} true if successful
+ * @throws if fails
+ */
+export const useDeleteApplication = (applicationId: string): UseQueryResult<boolean> => {
+    return useQuery('deleteApplication', async () => {
+        const { data, status } = await fetcher.delete<Failure>(`/application/${applicationId}`);
```
Comment (bwormser) [inline client/src/api/routes/applications.ts:200]: Should these fetcher.delete functions has an alternative typing other than failure like the functions above?

File: client/src/api/routes/users.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+/**
+ * @description POST /users/signIn with email & password
+ * @param {string} email 
+ * @param {string} password 
+ * @returns {UseQueryResult<boolean>} true on success, throws if fails
+ */
+export const useSignInWithEmail = (email: string, password: string): UseQueryResult<boolean> => {
+    return useQuery('signInWithEmail', async () => {
+        await doEmailSignIn(email, password);
+        const { data, status } = await fetcher.post<Failure>('/users/signIn');
```
Comment (bwormser) [inline client/src/api/routes/users.ts:15]: Same as my other comment, I'm not sure on the behavior of this, but wouldn't typescript throw an error if it doesn't always return failure?

File: client/src/api/routes/applications.ts (hunk 1 above)
Comment (mpolimen) [reply client/src/api/routes/applications.ts]: Yeah he def made a mistake; good eye

**Review by mpolimen — COMMENTED** [review]
I think it looks good, we can individually edit these when we use them in our own frontend features anyway

File: client/src/api/routes/applications.ts (hunk 2 above)
Comment (roccopolimen) [reply client/src/api/routes/applications.ts:200]: nah they only send a message on success so the Failure type covers  it. Can see why you are confused, but it works 👍

File: client/src/api/routes/applications.ts (hunk 1 above)
Comment (roccopolimen) [reply client/src/api/routes/applications.ts]: will change

**Pull request merged**